server/users/tests/routes/user-data.test.ts

```diff
@@ -1,57 +1,121 @@
+  test("Prevents unauthorized access if no user logged in", async () => {
     const res = await request(server.app).get(`/@me`);
-
     expect(res.status).toBe(401);
   });
 
-  test("should return user data (logged in)", async () => {
-    setLoggedInUser(BOBATAN_USER_ID);
+  test("returns data logged in user (cached)", async function () {
+    const cachedData = {
+      avatar_url: "/this_was_cached.png",
+      username: "super_cached",
+    };
+    mocked(cache().hGet).mockResolvedValueOnce(stringify(cachedData));
+    setLoggedInUser(JERSEY_DEVIL_USER_ID);
+
     const res = await request(server.app).get(`/@me`);
 
     expect(res.status).toBe(200);
-    expect(res.body).toEqual(BOBATAN_USER_DATA);
+    expect(res.body).toEqual(cachedData);
+    expect(cache().hGet).toBeCalledTimes(1);
+    expect(cache().hGet).toBeCalledWith(CacheKeys.USER, JERSEY_DEVIL_USER_ID);
   });
 
-  test("should return pinned boards for current realm", async () => {
-    setLoggedInUser(BOBATAN_USER_ID);
-    const res = await request(server.app).get(
-      `/@me/pins/realms/${TWISTED_MINDS_REALM_EXTERNAL_ID}`
+  test("caches logged in user data", async function () {
+    setLoggedInUser(JERSEY_DEVIL_USER_ID);
+
+    const res = await request(server.app).get(`/@me`);
+    expect(res.status).toBe(200);
+    expect(cache().hSet).toBeCalledTimes(1);
+    expect(cache().hSet).toBeCalledWith(
+      CacheKeys.USER,
+      JERSEY_DEVIL_USER_ID,
+      // TODO: this will fail if we change what we cache. We should not
+      // rely on the whole response being cached, but declare the object ourselves.
+      stringify(res.body)
     );
+  });
 
+  test("Returns data for the logged in user", async () => {
+    setLoggedInUser(JERSEY_DEVIL_USER_ID);
+    const res = await request(server.app).get(`/@me`);
     expect(res.status).toBe(200);
-    expect(res.body).toEqual(BOBATAN_PINNED_BOARDS);
+    expect(res.body).toEqual({
+      avatar_url: "/hannibal.png",
+      username: "jersey_devil_69",
+    });
   });
 
-  test("should only return pinned boards for current realm", async () => {
-    setLoggedInUser(BOBATAN_USER_ID);
-    const res = await request(server.app).get(
-      `/@me/pins/realms/${UWU_REALM_EXTERNAL_ID}`
-    );
+  const testUserPatch = {
+    username: "SnazzyNewName",
+    avatarUrl:
+      "https://firebasestorage.googleapis.com/v0/b/bobaboard-fb.appspot.com/o/images%2Fusers%2Favatar%2F4f5fde75-e670-4bb2-a04f-e90a5038ee34.jpeg?alt=media&token=7bdc5534-3159-4856-8d98-937cc1ee642f",
+  };
 
-    expect(res.status).toBe(200);
-    expect(res.body.pinned_boards).toEqual({});
+  test("prevents unauthorized PATCH access to the @me endpoint", async () => {
+    await wrapWithTransaction(async () => {
+      const res = await request(server.app).patch(`/@me`).send(testUserPatch);
+
+      expect(res.status).toBe(401);
+    });
   });
 
-  // TODO: check if any test user has outdated notifications or none, and test there
```

Comment:
this same TODO exists in the notifications tests, figured it was okay to remove

Reply:
thank you!